src/widgetastic_patternfly/__init__.py

```diff
@@ -2274,3 +2282,219 @@ class AggregateStatusMiniCard(AggregateStatusCard):
         'and h2[contains(@class, "card-pf-title")]'
         '//span[normalize-space(following::text())={@name|quote}]]'
     )
+
+
+class SparkLineChart(Widget, ClickableMixin):
+    """Represents the Spark Line Chart from Patternfly (Data Visualization).
+
+    Args:
+        id: id of SparkLineChart
+        locator: If id not applies, you can also supply a full locator
+
+    .. code-block:: python
+        spark_line_chart = SparkLineChart(id="sparklineChart")
+    """
+    ROOT = ParametrizedLocator("{@locator}")
+    BASE_LOCATOR = ".//div[@id={}]"
+    RECT = ".//*[contains(@class, 'c3-event-rects c3-event-rects-single')]//*"
+    tooltip = Text(".//div[contains(@class,'c3-tooltip-container')]")
+
+    def __init__(self, parent, id=None, locator=None, logger=None):
+        """Create the widget"""
+        Widget.__init__(self, parent, logger=logger)
+        if id:
+            self.locator = self.BASE_LOCATOR.format(quote(id))
+        elif locator:
+            self.locator = locator
+        else:
+            raise TypeError("You need to specify either id or locator")
+
+    def read(self, *args, **kwargs):
```

Comment:
I wonder if in the future we want to move towards using more of a generator approach for reads like this, allowing people to do a performance friendly search operation. This is NOT a requirement for this PR however.

Reply:
@digitronik why do you need these arguments?